test/test/custom_error.cpp

```diff
@@ -102,10 +102,21 @@ TEST_CASE("custom_error_logger")
     // Validate that handler translated exception
     REQUIRE_THROWS_AS(check_hresult(0x80000018), hresult_illegal_delegate_assignment);
     REQUIRE(s_loggerCalled);
+#ifndef __cpp_lib_source_location
     // In C++17 these fields cannot be filled in so they are expected to be empty.
     REQUIRE(s_loggerArgs.lineNumber == 0);
     REQUIRE(s_loggerArgs.fileName == nullptr);
     REQUIRE(s_loggerArgs.functionName == nullptr);
+#else
+    // GCC can only compile these tests in C++20 so these fields will be filled in.
```

Comment:
I don't quite understand these changes.  This test file is supposed to be compiled in C++17 mode so the source information should not be available.

Reply:
Using cppwinrt headers requires coroutine support to be available, and the experimental coroutine support from MSVC only works with... MSVC. For mingw-w64 targets (which use libc++ or libstdc++ depending on the toolchain) I am forced to compile everything in C++20; C++17 doesn't work.